### PR TITLE
[Core] Improve `ComputeNodalGradientProcess`, adding python test and missing documentation

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

nodal_gradient

Improve `ComputeNodalGradientProcess`:
- Missing constructor for model+parameters (standard one)
- Reorganize source file (cpp) more consistently with other processes.
- Adding some missing documentation
- Better dimension handler
- Improve defaults
- Missing IO
- Optimize includes

Adding python test, which allows to test both in serial and **MPI**

Adding missing documentation in `docs/pages/Kratos/Processes/Calculation/compute_nodal_gradient_process.md`.

**🆕 Changelog**

- [Improve `ComputeNodalGradientProcess`, adding python test and missing documentation](https://github.com/KratosMultiphysics/Kratos/commit/02bdf2aff14cbce1f952e3127fe2b5cecf81c77b)
- [Typo](https://github.com/KratosMultiphysics/Kratos/commit/945958ab0b77a728f16bbfc29dc22f768fbdb888)
- [Adding documentation](https://github.com/KratosMultiphysics/Kratos/commit/9f5cecf3b1def53c38fd5d35e9f8a451cf11d601)
- [Expose new constructors](https://github.com/KratosMultiphysics/Kratos/commit/110257c709890cfb0efa7148913df7edec9a6ec3)
- [Adding python test](https://github.com/KratosMultiphysics/Kratos/commit/d9e9284cb723b3e5a399f6805789164998181057)

### CODE REVIEW

**Comment by loumalouomega** [comment]
Anyone?

**Comment by ddiezrod** [comment]
@loumalouomega It seems there are many changes mixed here. Also, I dont understand what the image has to do with the gradient calculation : )

**Comment by loumalouomega** [comment]
> @loumalouomega It seems there are many changes mixed here. Also, I dont understand what the image has to do with the gradient calculation : )

Sorry, lately, Github is doing strange stuff when merging master

**Comment by loumalouomega** [comment]
> @loumalouomega It seems there are many changes mixed here. 

Now

> Also, I dont understand what the image has to do with the gradient calculation : )

The image is the gradient of the distance to the skin

File: kratos/processes/compute_nodal_gradient_process.h [hunk 1]
```diff
@@ -115,45 +155,66 @@ struct VariableVectorRetriever
...
+     * @param rGradientVariable (optional) The variable to store the computed gradient.
+     * @param rAreaVariable (optional) The variable representing nodal area for weighted gradient computation. Default is NODAL_AREA.
+     * @param NonHistoricalVariable (optional) Flag indicating if the origin variable is non-historical. Default is false.
+     */
     ComputeNodalGradientProcess(
         ModelPart& rModelPart,
-        const Variable<double>& rOriginVariable,
-        const Variable<array_1d<double,3> >& rGradientVariable,
+        const Variable<double>& rOriginVariable = DISTANCE,
+        const Variable<array_1d<double,3>>& rGradientVariable = DISTANCE_GRADIENT,
```
Comment (ddiezrod) [inline kratos/processes/compute_nodal_gradient_process.h]: Honestly I dont see why DISTANCE should be assumed to be the default variables

File: kratos/processes/compute_nodal_gradient_process.cpp [hunk 2]
```diff
@@ -328,14 +366,32 @@ void ComputeNodalGradientProcess<THistorical>::SynchronizeGradientAndVolume()
...
+    Parameters default_parameters = Parameters(R"(
+    {
+        "model_part_name"                : "PLEASE_DEFINE_A_MODEL_PART_NAME",
+        "origin_variable"                : "DISTANCE",
+        "gradient_variable"              : "DISTANCE_GRADIENT",
+        "area_variable"                  : "NODAL_AREA",
+        "non_historical_origin_variable" :  false
+    })" );
+
+    return default_parameters;
```
Comment (ddiezrod) [inline kratos/processes/compute_nodal_gradient_process.cpp]:
```suggestion
    return Parameters(R"(
    {
        "model_part_name"                : "PLEASE_DEFINE_A_MODEL_PART_NAME",
        "origin_variable"                : "DISTANCE",
        "gradient_variable"              : "DISTANCE_GRADIENT",
        "area_variable"                  : "NODAL_AREA",
        "non_historical_origin_variable" :  false
    })" );
```

File: kratos/processes/compute_nodal_gradient_process.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/processes/compute_nodal_gradient_process.h]: No one is forced to use it, but we ca simply removed if you want

**Review by ddiezrod — APPROVED** [review]

**Pull request merged**